src/app/utils/query-url-sanitization.ts

```diff
@@ -15,74 +17,44 @@ const guidRegex =
  /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
 
 
-export function normalizeQueryUrl(incomingUrl: string) {
-  incomingUrl = decodeURIComponent(incomingUrl);
+export function sanitizeQueryUrl(url: string): string {
+  url = decodeURIComponent(url);
 
   // Extract query string
-  let queryString: string = '';
-  const urlParts = incomingUrl.split('?');
-  if (urlParts.length > 1) {
-    const startOfQueryString = incomingUrl.indexOf('?');
-    queryString = incomingUrl.substring(startOfQueryString, incomingUrl.length - startOfQueryString);
-    queryString = processQueryParams(queryString.slice(1));
-  }
+  const { search } = parseSampleUrl(url);
+  const queryString: string = sanitizeQueryParameters(search);
 
   // Drop query string
-  incomingUrl = urlParts[0];
-
-  // Drop casts that use the /$/ pattern
-  incomingUrl = incomingUrl.split('/$/')[0];
+  let sanitizedUrl = url.split('?')[0];
 
   // Normalize parameters in param=<arbitraryKey> format with param={value} format
-  incomingUrl = incomingUrl.replace(functionParamInitialRegex, '<value>');
-  incomingUrl = incomingUrl.replace(functionParamFinalRegex, '<value>');
+  sanitizedUrl = sanitizedUrl.replace(functionParamInitialRegex, '<value>');
```

Comment:
Is it possible for us to get mention the resource in the generated arbitrary key?
e.g for `/me/messages/abcd` we can replace it with `/me/messages/{message-id}`

The reason for this is I'd like to reuse this in the autocomplete work